File: rtcbuilder/jp.go.aist.rtm.rtcbuilder.csharp/resource/AIST6/bar.cs
using System;
using System.Collections.Generic;
using System.Text;

using omg.org.RTC;
using RTC;
using org.omg.SDOPackage;



namespace bar
{

    public class bar
        : DataFlowComponentBase
    {
        private TimedShort m_InP1InData = new TimedShort();
        private InPort<TimedShort, RingBuffer<TimedShort>> m_InP1InPort;
        private TimedLong m_InP2InData = new TimedLong();
        private InPort<TimedLong, RingBuffer<TimedLong>> m_InP2InPort;
        private TimedDouble m_OutP1OutData = new TimedDouble();
        private OutPort<TimedDouble, RingBuffer<TimedDouble>> m_OutP1OutPort;
        private TimedFloat m_OutP2OutData = new TimedFloat();
        private OutPort<TimedFloat, RingBuffer<TimedFloat>> m_OutP2OutPort;

        private testImpl m_accProv = new testImpl();
        private CorbaPort m_testProvPort = new CorbaPort("accProv");



        public bar(Manager manager)
            : base(manager)
        {
            m_InP1InPort = new InPort<TimedShort, RingBuffer<TimedShort>>("InP1", m_InP1InData);
            registerInPort("InP1", m_InP1InPort);
            m_InP2InPort = new InPort<TimedLong, RingBuffer<TimedLong>>("InP2", m_InP2InData);
            registerInPort("InP2", m_InP2InPort);
            m_OutP1OutPort = new OutPort<TimedDouble, RingBuffer<TimedDouble>>("OutP1", m_OutP1OutData);
            registerOutPort("OutP1", m_OutP1OutPort);
            m_OutP2OutPort = new OutPort<TimedFloat, RingBuffer<TimedFloat>>("OutP2", m_OutP2OutData);
            registerOutPort("OutP2", m_OutP2OutPort);

            m_testProvPort.registerProvider("acc", "test", m_accProv);
            registerPort(m_testProvPort);

        }

        protected override ReturnCode_t onInitialize()
        {

            return ReturnCode_t.RTC_OK;
        }

        //protected override ReturnCode_t onFinalize()
        //{
        //    return ReturnCode_t.RTC_OK;
        //}

        //protected override ReturnCode_t onStartup(int ec_id)
        //{
        //    return ReturnCode_t.RTC_OK;
        //}

        //protected override ReturnCode_t onShutdown(int ec_id)
        //{
        //    return ReturnCode_t.RTC_OK;
        //}

        //protected override ReturnCode_t onActivated(int ec_id)
        //{
        //    return ReturnCode_t.RTC_OK;
        //}

        //protected override ReturnCode_t onDeactivated(int ec_id)
        //{
        //    return ReturnCode_t.RTC_OK;
        //}

        //protected override ReturnCode_t onExecute(int ec_id)
        //{
        //    return ReturnCode_t.RTC_OK;
        //}

        //protected override ReturnCode_t onAborting(int ec_id)
        //{
        //    return ReturnCode_t.RTC_OK;
        //}

        //protected override ReturnCode_t onError(int ec_id)
        //{
        //    return ReturnCode_t.RTC_OK;
        //}

        //protected override ReturnCode_t onReset(int ec_id)
        //{
        //    return ReturnCode_t.RTC_OK;
        //}

        //protected override ReturnCode_t onStateUpdate(int ec_id)
        //{
        //    return ReturnCode_t.RTC_OK;
        //}

        //protected override ReturnCode_t onRateChanged(int ec_id)
        //{
        //    return ReturnCode_t.RTC_OK;
        //}
    }

    public class barInit
    {
        public static string[] spec = new string[]
        {
            "implementation_id", "bar",
            "type_name",         "bar",
            "description",       "bartest",
            "version",           "2.0",
            "vendor",            "Tec",
            "category",          "same",
            "activity_type",     "SPORADIC",
            "max_instance",      "10",
            "language",          "C#",
            "lang_type",         "COMPILE"
        };

        public barInit(Manager manager)
        {
            Properties profile = new Properties(spec);
            manager.registerFactory(profile,
                delegate(Manager mgr) { return new bar(mgr); },
                delegate(RTObject_impl rtc) { rtc.Dispose(); });
        }
    }
}

File: rtcbuilder/jp.go.aist.rtm.rtcbuilder.vbdotnet/resource/AIST6/bar.vb
Imports RTC
Imports omg.org.RTC



Public Class bar
    Inherits DataFlowComponentBase

    Private m_InP1InData As TimedShort = New TimedShort()
    Private m_InP1InPort As InPort(Of TimedShort, RingBuffer(Of TimedShort))
    Private m_InP2InData As TimedLong = New TimedLong()
    Private m_InP2InPort As InPort(Of TimedLong, RingBuffer(Of TimedLong))
    Private m_OutP1OutData As TimedDouble = New TimedDouble()
    Private m_OutP1OutPort As OutPort(Of TimedDouble, RingBuffer(Of TimedDouble))
    Private m_OutP2OutData As TimedFloat = New TimedFloat()
    Private m_OutP2OutPort As OutPort(Of TimedFloat, RingBuffer(Of TimedFloat))

    Private m_accProv As testImpl  = New testImpl()
    Private m_testProvPort As CorbaPort = New CorbaPort("accProv")



    Public Sub New(ByVal manager As Manager)
        MyBase.New(manager)
        m_InP1InPort = New InPort(Of TimedShort, RingBuffer(Of TimedShort))("InP1", m_InP1InData)
        registerInPort("InP1", m_InP1InPort)
        m_InP2InPort = New InPort(Of TimedLong, RingBuffer(Of TimedLong))("InP2", m_InP2InData)
        registerInPort("InP2", m_InP2InPort)
        m_OutP1OutPort = New OutPort(Of TimedDouble, RingBuffer(Of TimedDouble))("OutP1", m_OutP1OutData)
        registerOutPort("OutP1", m_OutP1OutPort)
        m_OutP2OutPort = New OutPort(Of TimedFloat, RingBuffer(Of TimedFloat))("OutP2", m_OutP2OutData)
        registerOutPort("OutP2", m_OutP2OutPort)

        m_testProvPort.registerProvider("acc", "test", m_accProv)
        registerPort(m_testProvPort)

    End Sub

    Protected Overrides Function onInitialize() As ReturnCode_t

        Return ReturnCode_t.RTC_OK
    End Function

    'Protected Overrides Function onFinalize() As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function

    'Protected Overrides Function onStartup(ByVal ec_id As Integer) As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function

    'Protected Overrides Function onShutdown(ByVal ec_id As Integer) As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function

    'Protected Overrides Function onActivated(ByVal ec_id As Integer) As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function

    'Protected Overrides Function onDeactivated(ByVal ec_id As Integer) As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function

    'Protected Overrides Function onExecute(ByVal ec_id As Integer) As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function

    'Protected Overrides Function onAborting(ByVal ec_id As Integer) As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function

    'Protected Overrides Function onError(ByVal ec_id As Integer) As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function

    'Protected Overrides Function onReset(ByVal ec_id As Integer) As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function

    'Protected Overrides Function onStateUpdate(ByVal ec_id As Integer) As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function

    'Protected Overrides Function onRateChanged(ByVal ec_id As Integer) As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function
End Class

Public Class barInit

    Private spec() As String = New String() _
    { _
        "implementation_id", "bar", _
        "type_name",         "bar", _
        "description",       "bartest", _
        "version",           "2.0", _
        "vendor",            "Tec", _
        "category",          "same", _
        "activity_type",     "PERIODIC", _
        "max_instance",      "10", _
        "language",          "Visual Basic", _
        "lang_type",         "COMPILE" _
    }

    Public Sub New(ByVal manager As Manager)
        Dim profile As Properties = New Properties(spec)
        manager.registerFactory(profile, _
            New RtcNewFunc(AddressOf Createbar), _
            New RtcDeleteFunc(AddressOf Deletebar))
    End Sub

    Public Function Createbar(ByVal manager As Manager) As RTObject_impl
        Return New bar(manager)
    End Function

    Public Sub Deletebar(ByVal rtc As RTObject_impl)
        rtc.Dispose()
    End Sub

End Class
